hoomd/pytest/test_dcd.py

```diff
@@ -0,0 +1,40 @@
+import hoomd
+import pytest
+import numpy as np
+
+
+def test_attach(simulation_factory, two_particle_snapshot_factory, tmp_path):
+    filename = tmp_path / "temporary_test_file.dcd"
+    sim = simulation_factory(two_particle_snapshot_factory())
+    dcd_dump = hoomd.write.DCD(filename, hoomd.trigger.Periodic(1))
+    sim.operations.add(dcd_dump)
+    sim.run(10)
+
+
+def test_write(simulation_factory, two_particle_snapshot_factory, tmp_path):
+    garnett = pytest.importorskip("garnett")
+    dcd_reader = garnett.reader.DCDFileReader()
+    filename = tmp_path / "temporary_test_file.dcd"
+    sim = simulation_factory(two_particle_snapshot_factory())
+    dcd_dump = hoomd.write.DCD(filename, hoomd.trigger.Periodic(1))
+    sim.operations.add(dcd_dump)
+    snap = sim.state.snapshot
+    positions = []
+    for i in range(10):
+        snap = sim.state.snapshot
+        position1 = np.asarray(snap.particles.position[0])
+        position2 = np.asarray(snap.particles.position[1])
+        position1 += 0.1 * i * (-1)**i
+        position2 += 0.1 * (i + 1) * (-1)**(i - 1)
+        if snap.exists:
```

Comment:
Move this `if snap.exists` up to include all code that accesses snap.particles.position.